tools/serverpod_cli/lib/src/analyzer/dart/definition_analyzers/method_analyzer.dart

```diff
@@ -0,0 +1,132 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/code_analysis_collector.dart';
+import 'package:serverpod_cli/src/analyzer/dart/definition_analyzers/parameter_analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/dart/definitions.dart';
+import 'package:source_span/source_span.dart';
+
+const _excludedMethodNameSet = {
+  'streamOpened',
+  'streamClosed',
+  'handleStreamMessage',
+  'sendStreamMessage',
+  'setUserObject',
+  'getUserObject',
+};
+
+abstract class MethodAnalyzer {
+  static MethodDefinition? analyze(
+    MethodElement method,
+    Parameters parameters,
+    CodeAnalysisCollector collector,
+  ) {
+    if (!_hasValidReturnType(
+      dartType: method.returnType,
+      dartElement: method,
+      collector: collector,
+    )) {
```

Comment:
Suggestion: Change the name of the method to `validateReturnType` as this is not a strict bool check we are doing. It could be cleaner to simply return a list with errors and deal with where they go at this level. 

With the current pattern, we are mutating the object deep inside the logic which I think would be nice to get away from. My reasoning here is that in the longterm the validation logic will start to spread around in the code and it may be hard to track down where we are doing a specific validation.

Reply:
Great suggestion, will implement it. Better to try and have the same abstraction levels.